tests/pass.rs

```diff
@@ -0,0 +1,46 @@
+use nickel::program::Program;
+use nickel::term::Term;
+use std::ffi::{OsString, OsStr};
+use std::fs::read_dir;
+use std::path::PathBuf;
+use std::thread;
+
+// By default, tests are run with 2MB stack size, which can overflow in debug mode. We run the
+// tests with an increased stack size.
+const STACK_SIZE: usize = 4 * 1024 * 1024;
+
+fn run(path: impl Into<OsString>) {
+    let path = path.into();
+    let mut p = Program::new_from_file(path.clone()).expect("could not load file as a program");
+    assert_eq!(
+        p.eval(),
+        Ok(Term::Bool(true)),
+        "error evaluating {}",
+        path.to_string_lossy(),
+    );
+}
+
+#[test]
+fn tests() {
+    let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    d.push("tests/pass");
+
+    let children: Vec<_> = read_dir(d)
+        .unwrap()
+        .into_iter()
+        .map(|file| {
+            let file = file.unwrap();
+            let path = file.path();
+            let name = String::from(path.file_name().map(OsStr::to_string_lossy).unwrap_or(path.to_string_lossy()));
+            thread::Builder::new()
```

Comment:
Does that mean that all the tests will run in parallel or is there some kind of implicit limit?

Reply:
Unless this is too tricky to do, I think it would be much nicer to have one test per input file, so that the test runner can handle the parallelism and all the resource management/error reporting itself

Reply:
> Does that mean that all the tests will run in parallel or is there some kind of implicit limit?
Yes, It should spawn one thread per input file

> Unless this is too tricky to do, I think it would be much nicer to have one test per input file, so that the test runner can handle the parallelism and all the resource management/error reporting itself

Since we need to spawn a thread manually to use a bigger stack size anyway, I went with doing everything in one function.  But I think you're right, it's cleaner with one function per test. It just means that we have to write a new function when we add a new test file, but this sounds like a reasonable expectation.